src/Backend/DataSeriesFactory.js

```diff
@@ -0,0 +1,29 @@
+export default class TraceFactory {
+    static getXValuesByPoints(min_energy, max_energy, pointNumber) {
+        const interval = (max_energy - min_energy) / (pointNumber - 1)
```

Comment:
Imagine user requests a single point. Then you will have division by zero.
Simple solution for that is to check if number of points is == 1 and return then (max + min)/2.